modules/ducktests/tests/ignitetest/tests/two_phased_rebalanced_test.py

```diff
@@ -0,0 +1,278 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module contains Cellular Affinity tests.
+"""
+from typing import List
+from ducktape import errors
+from ducktape.cluster.cluster import ClusterNode
+from ducktape.mark.resource import cluster
+from ignitetest.services.utils.ignite_aware import IgniteAwareService
+
+from ignitetest.services.utils.ignite_configuration.cache import CacheConfiguration, Affinity
+
+from ignitetest.services.ignite import IgniteService
+from ignitetest.services.ignite_app import IgniteApplicationService
+from ignitetest.services.utils.control_utility import ControlUtility
+from ignitetest.services.utils.ignite_configuration import IgniteConfiguration, DataStorageConfiguration
+from ignitetest.services.utils.ignite_configuration.data_storage import DataRegionConfiguration
+from ignitetest.services.utils.ignite_configuration.discovery import from_ignite_cluster
+from ignitetest.services.utils.util import copy_file_to_dest
+from ignitetest.utils import ignite_versions
+from ignitetest.utils.ignite_test import IgniteTest
+from ignitetest.utils.version import IgniteVersion, DEV_BRANCH, LATEST_2_9, LATEST_2_8
+
+NUM_NODES_CELL = 4
+
+NUM_CELL = 2
+
+ATTRIBUTE = "CELL"
+
+CACHE_NAME = "test-cache"
+
+
+# pylint: disable=W0223
+class TwoPhasedRebalancedTest(IgniteTest):
+    """
+    Two-phase rebalancing test case.
+    """
+    # pylint: disable=R0914
+    @cluster(num_nodes=(NUM_NODES_CELL * NUM_CELL) + 1)
+    @ignite_versions(str(DEV_BRANCH), str(LATEST_2_9), str(LATEST_2_8))
+    def two_phased_rebalancing_test(self, ignite_version):
```

Comment:
WDYT if we add some tests that check concurrent cases? Smth like rebalance while transaction with batch update/delete is open on other node in a cell? Is it a valid case to check it there?

Reply:
Agreed that there is no other cases.